spec/requests/widgets_spec.rb

```diff
@@ -62,43 +62,83 @@
+
+          expect(miq_widget.miq_widget_contents.count).to eq(0)
+          post(api_widget_url(nil, miq_widget), :params => gen_request(:generate_content))
+          expect(response).to have_http_status(:ok)
+          expect(miq_widget.miq_widget_contents.count).to eq(1)
+        end
+
+        it "multiple widgets" do
+          first_miq_widget = FactoryBot.create(:miq_widget, :visibility => {:roles => '_ALL_'}, :resource => MiqReport.first)
+          second_miq_widget = FactoryBot.create(:miq_widget, :visibility => {:roles => '_ALL_'}, :resource => MiqReport.first)
+
+          ws = FactoryBot.create(:miq_widget_set, :name => "Home", :userid => @user1.userid, :group_id => @group1.id)
+          first_miq_widget.make_memberof(ws)
+          second_miq_widget.make_memberof(ws)
+          api_basic_authorize('widget_generate_content')
+
+          post(api_widgets_url, :params => gen_request(:generate_content, [{"href" => api_widget_url(nil, first_miq_widget)}, {"href" => api_widget_url(nil, second_miq_widget)}]))
+
+          expect(response).to have_http_status(:ok)
+          expect(first_miq_widget.miq_widget_contents.count).to eq(1)
+          expect(second_miq_widget.miq_widget_contents.count).to eq(1)
+        end
       end
 
-      it "generate_content of multiple Widgets" do
-        first_miq_widget = FactoryBot.create(:miq_widget)
-        second_miq_widget = FactoryBot.create(:miq_widget)
-        api_basic_authorize('widget_generate_content')
-        expect(MiqTask.count).to eq(0)
-
-        post(api_widgets_url, :params => gen_request(:generate_content, [{"href" => api_widget_url(nil, first_miq_widget)}, {"href" => api_widget_url(nil, second_miq_widget)}]))
-
-        expected = {
-          "results" => a_collection_containing_exactly(
-            a_hash_including(
-              "message" => a_string_matching(/#{first_miq_widget.id}.* content generation/i),
-              "success" => true,
-              "href"    => api_widget_url(nil, first_miq_widget)
-            ),
-            a_hash_including(
-              "message" => a_string_matching(/#{second_miq_widget.id}.* content generation/i),
-              "success" => true,
-              "href"    => api_widget_url(nil, second_miq_widget)
-            )
-          )
-        }
-        expect(response.parsed_body).to include(expected)
-        expect(response).to have_http_status(:ok)
-        expect(MiqTask.count).to eq(2)
+      context "generate_content for a user" do
+        before do
+          MiqReport.seed_report("Vendor and Guest OS")
+          feature1 = MiqProductFeature.find_all_by_identifier("dashboard_admin")
+          @user1   = FactoryBot.create(:user, :role => "role1", :features => feature1)
+          @group1  = @user1.current_group
+        end
+
+        it "single widget" do
+          miq_widget = FactoryBot.create(:miq_widget, :visibility => {:roles => '_ALL_'}, :resource => MiqReport.first)
```

Comment:
What about to move all `FactoryBot`s to `let`s and share them in those two `it`s ?